src/hooks/check_undeclared_software.py

```
@@ -78,15 +78,19 @@ def get_staged_files() -> list[str]:
-    command: list[str], check: bool = True
+    command: list[str], check: bool = True, debug: bool = False
 ) -> subprocess.CompletedProcess[str]:
     """Run a command safely and return the completed process.
 
     Args:
         command (list[str]): Command to run
         check (bool): Whether to check the return code of the command
+        debug (bool): Whether to log the command output
     """
-    return subprocess.run(command, check=check, capture_output=True, text=True)
+    result = subprocess.run(command, check=check, capture_output=True, text=True)
```

Ruff (0.14.1)

`subprocess` call: check for execution of untrusted input

(S603)